### ISSUE TITLE
Use NOAA and WWIS for weather information

### ISSUE THREAD

**Issue by aaronchantrill** [NaomiProject/Naomi#96]
I don't use Yahoo!. I'm not sure why Weather Underground is used to validate the location (zip code or major city) being sent to Yahoo! to get weather information.

I'd like to expand the standard weather plugin and ask the user to choose a weather service before asking for a location. I would start with NOAA for US weather and WWIS for international weather, then let others expand the list as needed for their own regional weather.

**Comment by aaronchantrill** [comment]
So now the Yahoo! api has been retired:
Important EOL Notice: As of Thursday, Jan. 3, 2019, the YQL service at query.yahooapis.com will be retired. This will impact users of datatables.org as well as developers who creates features using this YQL service. 
To continue using our free Yahoo Weather APIs, use https://weather-ydn-yql.media.yahoo.com/forecastrss. Follow instructions here to get credentials and onboard to this free Yahoo Weather API service./p>

**Comment by aaronchantrill** [comment]
For WWIS, it looks like you just need these two calls:
https://worldweather.wmo.int/{lang}/json/Country_{lang}.xml
to find the closest city with a forecast, and
https://worldweather.wmo.int/{lang}/json/{cityId}_{lang}.xml
I'm not sure why the language is here twice.

Here is a simple implementation:
```python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import requests
import json

url = "https://worldweather.wmo.int/en/json/Country_en.xml"
print( "Requesting url {}".format(url) )
response = requests.get(url)
print( "Request finished" )
jsondoc = str(response.content,'utf-8')

countrydata = json.loads(jsondoc)
countries = {}
for country in countrydata["member"]:
    if(type(countrydata["member"][country]) is dict):
        memId = countrydata["member"][country]["memId"]
        memName = countrydata["member"][country]["memName"]
        countries[memName]=country
print("Select country from list:")
for country in sorted(countries):
    print("{}".format(country))

country=""
while country not in countries:
    country = input("Select Country: ")
countryid = countries[country]

regions = {}
country_contains_regions = False
# All we have at this point is a list of all the cities
# in the country. These are divided into strings of
# "City, Region" or just "City" if the city is not part
# of a larger region.
# Since I don't want to list every city in the United
# States and ask the user to locate their city in a big
# list, but rather have the user select the state and
# then the city, I want to make a list of all the states
# and then let the user select from a list of cities in
# that state. However, that creates a problem if there
# is a country that has cities that are part of a larger
# region and other cities that are not.
for city in countrydata["member"][countryid]["city"]:
    if("," in city["cityName"]):
        cityId = city["cityId"]
        cityName,regionName = city["cityName"].split(", ")
        if regionName not in regions.keys():
            regions[regionName]={}
        regions[regionName][cityName]=cityId
        country_contains_regions = True
    else:
        regions[city["cityName"]]=city["cityId"]
if(not regions):
    print("Weather information is not available in {}.".format(country))
    print("Please check nearby cities in other countries")
else:
    print("")
    if(country_contains_regions):
        print("Please select your region:")
        for region in sorted(regions):
            print("{}".format(region))
        regionName = ""
        while regionName not in regions:
            regionName = input("Select region: ")
        print("")
        if(type(regions[regionName]) is dict):
            cities=regions[regionName]
            print("Select a city from the list:")
            for city in regions[regionName]:
                print("{}".format(city))
            cityName = ""
            while cityName not in regions[regionName]:
                cityName = input("Select City: ")
            cityId = regions[regionName][cityName]
            print("")
        else:
            cityName = regionName
            cityId = regions[regionName]
    else:
        # in this case these are cities not regions
        for city in sorted(regions):
            print("{}".format(city))
        cityName = ""
        while cityName not in regions:
            cityName = input("Select City: ")
        cityId = regions[cityName]
    url="https://worldweather.wmo.int/en/json/{}_en.xml".format(cityId)
    # print( "Requesting url {}".format(url) )
    response = requests.get(url)
    # print( "Request finished" )
    jsondoc = str(response.content,'utf-8')
    weatherdata = json.loads(jsondoc)
    # print(json.dumps(weatherdata, indent=4, sort_keys=True))
    forecast = {}
    for day in weatherdata["city"]["forecast"]["forecastDay"]:
        forecast[day["forecastDate"]]={}
        forecast[day["forecastDate"]]["weather"]=day["weather"]
        forecast[day["forecastDate"]]["high"]=day["maxTempF"]
        forecast[day["forecastDate"]]["low"]=day["minTempF"]
    if(not forecast):
        print("Sorry, forcast information is not currently available for {} in {}".format(weatherdata["city"]["cityName"],country))
    for day in sorted(forecast.keys()):
        print("{}\t{}\tLow: {}\tHigh: {}".format(day,forecast[day]["weather"].ljust(15),forecast[day]["low"],forecast[day]["high"]))
```

**Comment by aaronchantrill** [comment]
We are currently in the middle of a government shutdown in the U.S., and the NOAA site is not currently functioning. It is just forwarding people to forecast.weather.gov for current weather information. Just FYI.

**Comment by aaronchantrill** [comment]
As I am re-writing this, I will also incorporate issue #98 Move plugin specific settings into the plugin. This requires the user to drill down by country, region, and city to get to a forecast, so rather than replacing the zip code lookup code from populate.py, come up with a way of importing requested variables, whether they are required or not, and validation/generation process directly from the plugin.

**Comment by aaronchantrill** [comment]
Resolved in [PR#183 Replacement for weather plugin](https://github.com/NaomiProject/Naomi/pull/183)
